### ISSUE TITLE
parse_bar:162: malformed line:

### ISSUE THREAD

**Issue by TheChosenOne84** [vivien/i3blocks#348]
Hi,

first I would like to thank you for this awesome project.

I have some trouble on my fresh installed Ubuntu 18.04 using i3blocks 1.4

I copied the config file from my old pc using scp. If I execute i3blocks i get errors like

ERROR parse_bar:162: malformed line: IFACE=eno1
ERROR main:70: Try 'i3blocks -h' for more information.

or

ERROR parse_bar:162: malformed line: DATEFMT=+%d.%m.%Y %H:%M:%S
ERROR main:70: Try 'i3blocks -h' for more information.

My first idea was problems with the font-awesome icons, but as you can see above also lines without these special characters have problems. Do you know this problem or can give me some hints how to debug it?

Thank you!

**Comment by unixlab** [comment]
Hi,

based on the code in this repository ( I'm not sure if the debian maintainer applied patches ) the problem might be the capital letters:

```
switch (*line) {
/* Comment or empty line? */
case '#':
case '\0':
	break;

/* Section? */
case '[':
	/* Finalize previous block */
	if (block)
		block_setup(block);

	block = add_block(bar);
	if (!block)
		goto free;

	/* Init the block with default settings (if any) */
	memcpy(block, &global, sizeof(struct block));

	if (parse_section(line, block->default_props.name, sizeof(block->default_props.name)))
		goto free;

	bdebug(block, "new block");
	break;

/* Property? */
case 'a' ... 'z':
	if (!block) {
		debug("parsing global properties");
		block = &global;
	}

	if (parse_property(line, &block->default_props, false))
		goto free;

	break;

/* Syntax error */
default:
	error("malformated line: %s", line);
	goto free;
}
```

As you can see, there're cases for comments, empty lines, bracket and **lower** characters from a to z.

Regards
Felix

**Comment by kb100** [comment]
@unixlab The corresponding code in this repo is found at https://github.com/vivien/i3blocks/blob/7bcb556a7e59a2b43809e25e186f7d385f280976/ini.c#L48, which uses isalpha instead of 'a' ... 'z'. In order to use the instructions and blocks found in the git versions of i3blocks and i3blocks-contrib, one must use the git versions, not your distribution's version, as they are out of date.

**Comment by unixlab** [comment]
@kb100 Indeed, the code I mentioned doesn't exists in the master branch.
But @TheChosenOne84 specifically mentioned version 1.4, which uses 'a' ... 'z'.
I just wanted to explain the behavior in version 1.4 and a possible solution.
Maybe I should have made this more clear.

However, this isn't a code issue at all.
It's, as @kb100 pointed out, a problem with outdated packages in the Ubuntu repositories.

**Comment by kb100** [comment]
Indeed, in any case, thank you for pointing out that the ubuntu repositories are out of date, as this may help other ubuntu users to know that this is indeed the problem.

**Comment by TheChosenOne84** [comment]
Hi @unixlab and @kb100,
after removing the repositories version and compiling the code from github everything works perfect.

Thanks for your help!
